src/Abp.EntityFrameworkCore.EFPlus/EntityFrameworkCore/EFPlus/AbpEntityFrameworkCoreEFPlusExtensions.cs

```diff
@@ -0,0 +1,157 @@
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <typeparam name="TPrimaryKey">Primary key type</typeparam>
+        /// <param name="repository">Repository</param>
+        /// <param name="predicate">Predicate to filter entities</param>
+        /// <returns></returns>
+        public static async Task<int> BatchDeleteAsync<TEntity, TPrimaryKey>([NotNull] this IRepository<TEntity, TPrimaryKey> repository, Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<TPrimaryKey>
+        {
+            Check.NotNull(repository, nameof(repository));
+
+            var query = repository.GetAll().IgnoreQueryFilters();
+
+            var abpFilterExpression = GetFilterExpressionOrNull<TEntity, TPrimaryKey>(repository.GetIocResolver());
+            var filterExpression = ExpressionCombiner.Combine(predicate, abpFilterExpression);
+
+            if (filterExpression != null)
+            {
+                query = query.Where(filterExpression);
+            }
+
+            return await query.DeleteAsync();
+        }
+
+        /// <summary>
+        /// Deletes all matching entities permanently for given predicate
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <param name="repository">Repository</param>
+        /// <param name="predicate">Predicate to filter entities</param>
+        /// <returns></returns>
+        public static async Task<int> BatchDeleteAsync<TEntity>([NotNull] this IRepository<TEntity> repository, Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<int>
+        {
+            return await repository.BatchDeleteAsync<TEntity, int>(predicate);
+        }
+
+        /// <summary>
+        /// Updates all matching entities using given updateExpression for given predicate
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <typeparam name="TPrimaryKey">Primary key type</typeparam>
+        /// <param name="repository">Repository</param>
+        /// /// <param name="updateExpression">Update expression</param>
+        /// <param name="predicate">Predicate to filter entities</param>
+        /// <returns></returns>
+        public static async Task<int> BatchUpdateAsync<TEntity, TPrimaryKey>([NotNull]this IRepository<TEntity, TPrimaryKey> repository, Expression<Func<TEntity, TEntity>> updateExpression, Expression<Func<TEntity, bool>> predicate)
+            where TEntity : Entity<TPrimaryKey>
+        {
+            Check.NotNull(repository, nameof(repository));
+
+            var query = repository.GetAll().IgnoreQueryFilters();
+
+            var abpFilterExpression = GetFilterExpressionOrNull<TEntity, TPrimaryKey>(repository.GetIocResolver());
+            var filterExpression = ExpressionCombiner.Combine(predicate, abpFilterExpression);
+
+            if (filterExpression != null)
```

Comment:
filterExpression should never be null (assuming the predicate is not null). So throw exception instead of updating all table! This might be discussed, but I think we should not allow to update all table which can be dangerous.